bundles/org.openhab.binding.smarther/src/main/java/org/openhab/binding/smarther/internal/handler/SmartherBridgeHandler.java

```diff
@@ -0,0 +1,762 @@
+        updateApiCallsCounter();
+        return getSmartherApi().getModulePrograms(plantId, moduleId);
+    }
+
+    @Override
+    public boolean isAuthorized() {
+        try {
+            final AccessTokenResponse tokenResponse = getAccessTokenResponse();
+            onAccessTokenResponse(tokenResponse);
+
+            return (tokenResponse != null && tokenResponse.getAccessToken() != null
+                    && tokenResponse.getRefreshToken() != null);
+        } catch (SmartherAuthorizationException e) {
+            return false;
+        }
+    }
+
+    @Override
+    public boolean isOnline() {
+        return (thing.getStatus() == ThingStatus.ONLINE);
+    }
+
+    @Override
+    public String authorize(String redirectUrl, String reqCode, String notificationUrl)
+            throws SmartherGatewayException {
+        try {
+            logger.debug("Bridge[{}] Call API gateway to get access token. RedirectUri: {}", thing.getUID(),
+                    redirectUrl);
+
+            final OAuthClientService localOAuthService = this.oAuthService;
+            if (localOAuthService == null) {
+                throw new SmartherAuthorizationException("Authorization service is null");
+            }
+
+            // OAuth2 call to get access token from received authorization code
+            localOAuthService.getAccessTokenResponseByAuthorizationCode(reqCode, redirectUrl);
+
+            // Store the notification URL in bridge configuration
+            Configuration configuration = editConfiguration();
+            configuration.put(PROPERTY_NOTIFICATION_URL, notificationUrl);
+            updateConfiguration(configuration);
+            config.setNotificationUrl(notificationUrl);
+            logger.debug("Bridge[{}] Store notification URL: {}", thing.getUID(), notificationUrl);
+
+            // Reschedule the polling thread
+            schedulePoll();
+
+            return config.getClientId();
+        } catch (OAuthResponseException e) {
+            throw new SmartherAuthorizationException(e.toString(), e);
+        } catch (OAuthException | IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            throw new SmartherGatewayException(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public boolean equalsThingUID(String thingUID) {
+        return thing.getUID().getAsString().equals(thingUID);
+    }
+
+    @Override
+    public String formatAuthorizationUrl(String redirectUri) {
+        try {
+            final OAuthClientService localOAuthService = this.oAuthService;
+            if (localOAuthService != null) {
+                return localOAuthService.getAuthorizationUrl(redirectUri, null, thing.getUID().getAsString());
+            }
+        } catch (OAuthException e) {
+            logger.warn("Bridge[{}] Error constructing AuthorizationUrl: ", thing.getUID());
```

Comment:
Was the AuthorizationUrl supposed to be  included in this message?
You should also include the exception or at least the exception's message in the log as well.

Reply:
Yes, exception message need to be included. Thanks!